src/pages/register.tsx

```diff
@@ -0,0 +1,43 @@
+import { type NextPage } from "next";
+import { api } from "../utils/api";
+import { RegisterForm } from "../components/RegisterForm";
+import type { RegisterValues } from "../schemas/registerSchema";
+import { useToast } from "../hooks/useToast";
+import { Layout } from "../components/Layout";
+import { useAuth } from "../hooks/useAuth";
+import Router, { withRouter } from "next/router";
```

Comment:
Let's use `const router = useRouter();` instead of `import Router from "next/router";` I've never seen someone using `Router` object directly. Maybe it is okay. but let's go with a hook version